### ISSUE TITLE
[Feature Request] A Modern Web-UI / Just Build In Tamland's Alternative Web Frontend

### ISSUE THREAD

**Issue by grravity** [airsonic-advanced/airsonic-advanced#373]
## Problem description

The current Airsonic/Airsonic-advanced web-UI is very dated and un-intuitive.  File names do not display well, feels a bit slow overall.
I would suggest a web-ui overhaul, at the very least a more modern theme to select from if users which to use the old (current) theme still for whatever reason. 

https://github.com/tamland/airsonic-frontend
Is very very nice, and it would be nice to have this just built right in as a theme. Would save work to have to create one from scratch. If I could ask for any tweaks to it, it would be to see the codec & bitrate/samplerate for everything and currently playing. Would be nice to get this built in by the next stable release.

## System information

* **Airsonic version**: *10.6.0.20200424173747 – April 24, 2020 at 1:37:47 PM EDT*
 * **Operating system**: *Windows 10*
 * **Java version**: *AdoptOpenJDK - JRE-11.0.6.10*
 * **Proxy server**: *Caddy v2*
 * **Client**: *Firefox-ESR 68.12.*
 * **Language**: *American English*

**Comment by SteveDinn** [comment]
I can't +1 this enough. I've been using that UI and it's awesome. I only go back to the native UI to tweak the config or to rescan.

**Comment by SteveDinn** [comment]
@tamland

**Comment by randomnicode** [comment]
Airsonic-advanced and vanilla airsonic's web-UI may look alike, but behind the scenes they're actually not similar at all. The frontend was replaced while keeping the look the same (too many changes otherwise). Don't remember if the changes were pre 11.x (which you're on) or post.

Airsonic-Advanced's web UI actually uses websockets to converse with the server (Airsonic vanilla just used a pre alpha version of DWR).

Similarly Airsonic used obsolete (in HTML5) frames and framesets, which Airsonic-Advanced had to revamp and redo to get HTML5 compliant.

Regardless, there are a few proposals here:
- Add new themes: Theme additions may be done by anyone, please do add them! We've added the Dark Reader theme, for instance. If there is more demand, certain web elements may be made themeable
- Add Tamland's UI: This is more than a theming change, this requires adding a completely new UI framework, which can be done and we're not against, but needs some thought about how to coordinate the packaging of the two products together.
  - there needs to be a way to switch "engines" (even from the new one back to the old)
  - any possibility of using websockets instead of REST, @tamland?
  - licensing seems fairly compatible
  - initial thoughts might be to bundle up at release time from external project, and provide a way to switch in code to serve these files instead of those

**Comment by SteveDinn** [comment]
If the Airsonic-Advanced "native" UI was only the administration options and configuration, and clients were separate, I don't think that would necessarily be a bad thing. And it would encourage the results returned by the Airsonic API to be equivalent to what we see in the native UI. Currently, this is not always the case, which is [a bug that I've logged previously](https://github.com/airsonic-advanced/airsonic-advanced/issues/364).

**Comment by tamland** [comment]
I support the idea, but I think it's best to keep them separate. It's easy to run an additional container and 'building it in' doesn't really achieve anything. If you are to to do it, serving the bundle unmodified as static resources from airsonic is probably the easiest way (no idea if airsonic is going to conflict with that though).

To move forward it and eventually replace the current UI what we need is to improve and extend the API. A lot of things are simply impossible or hard to do right now due to missing API.

**Comment by grravity** [comment]
> I support the idea, but I think it's best to keep them separate. It's easy to run an additional container and 'building it in' doesn't really achieve anything. 

Possibly a stupid question, but this would only be the case if you are in fact running it in a container host such as Docker/podman, no?
For example a native Airsonic(advanced) install, or even a Windows install of Airsonic? Is it even possible to install your front-end in these cases? Especially the latter?

**Comment by tamland** [comment]
@grravity Sure I guess. But you can host it on any web server. Even opening index.html from a local folder in a browser will mostly work (as long as you don't refresh). There's also https://airsonic.netlify.app which you can use without the need to host it yourself.

**Comment by grravity** [comment]
> Even opening index.html from a local folder in a browser will mostly work

That kinda defeats the purpose imo.

>There's also https://airsonic.netlify.app which you can use without the need to host it yourself.

A good workaround, but it does slightly defeat the purpose as to why _some people_ self-host. 

>Sure I guess. But you can host it on any web server.

Understandable, but still it would be another thing users have to host versus it just "replacing" the old UI imo which would be a better option. If you ask me I think the default UI is one of the bigger things holding Airsonic back. It's great when logged in through Mopidy, or Ultrasonic or whatever third party client. The web-ui feels just painful to use imo.

A lot of the text doesn't even show in entirety unless you minimize the side bar, it's fairly slow a bit unresponsive.

I mean this is what people interface with (unless you never use the web-ui). I'm not against an older style, function over form look, but the current ui doesn't even feel very functional, never mind nice looking.

I feel airsonic advanced would be a good testing grounds for this.  I know you stated it could use a little work, I don't disagree there were a few things that stood out to me, and also it's missing a handful of pages like settings. But if it could be "built into" airsonic, however best, and allow users to switch between them, I think that would be a good start tbh. It would provide as a testing ground. 
Your frontend feels very responsive as far as actually navigating goes, that alone is an improvement over the current ui if you ask me.

It would be so much easier to just include it right in the airsonic builds, rather then needing to host an entirely different frontend alongside it.

**Comment by wabkia** [comment]
randomnicode already proposed an integration that will include the tamland ui with airsonic-advanced, and contacted key players to get things started.

> If you ask me I think the default UI is one of the bigger things holding Airsonic back.

airsonic or airsonic-advanced? holding airsonic back from what?

as a long time *sonic user i think the tamland ui needs quite a bit of work before its meeting the core functionality of the airsonic-advanced web gui, let alone feature parity. im not a fan of the look, and until i was forced off id be using the original ui.

**Comment by SteveDinn** [comment]
Yeah, having a choice of clients is good. After some thought on it, I'm personally fine with installing it as a separate thing. It's no different than having to install a client on my phone. I do think that the native client should be using the Airsonic API so that it can "eat its own dog food" so to speak. It's clear to me that this is currently not the case due to some behaviour differences I've noticed.
